packages/opds-web-client/src/components/DownloadButton.tsx

```diff
@@ -68,15 +68,20 @@ export default class DownloadButton extends React.Component<
   fulfill() {
     if (this.isIndirect()) {
       return this.props
-        .indirectFulfill(this.props.url, this.props.indirectType)
+        .indirectFulfill?.(this.props.url, this.props.indirectType)
         .then(url => {
           window.open(url, "_blank");
         });
     } else {
-      return this.props.fulfill(this.props.url).then(blob => {
+      return this.props.fulfill?.(this.props.url).then(blob => {
         download(
           blob,
-          generateFilename(this.props.title, this.fileExtension()),
+          generateFilename(
+            // the following is meant to smoothly handle the case where
+            // this.props.title is undefined or null
+            this.props.title ?? "untitled",
```

Comment:
I wasn't totally sure what to do in this situation. It was a case that previously could have happened but wasn't really being handled.

Reply:
I'm not sure if this is the safest option but seems okay for now.

Reply:
Yea I wasn't sure what to do here. Seemed like we should do something... in the past this would have caused a type error and the app would have crashed, so this is better than that at least.